components/ViewPortDetector.tsx

```diff
@@ -84,19 +90,21 @@ export const ViewPortDetector: React.FC<Props> = ({
           pageX: number,
           pageY: number
         ) => {
-          onChange(
-            checkInViewPort(
-              parentLayoutRef.current,
-              {
-                x: pageX,
-                y: pageY,
-                width,
-                height,
-              },
-              percentWidth,
-              percentHeight
-            )
+          const newValue = checkInViewPort(
+            parentLayoutRef.current,
+            {
+              x: pageX,
+              y: pageY,
+              width,
+              height,
+            },
+            percentWidth,
+            percentHeight
           );
+          if (newValue === true && runOnce) {
```

Comment:
I think we just need to check `if (newValue && runOnce)`

Reply:
All updated now 🔄